basyx/aas/model/provider.py

```diff
@@ -95,14 +104,157 @@ def get_identifiable(self, identifier: Identifier) -> _IT:
 
     def add(self, x: _IT) -> None:
         if x.id in self._backend and self._backend.get(x.id) is not x:
-            raise KeyError("Identifiable object with same id {} is already stored in this store"
-                           .format(x.id))
+            raise KeyError(
+                "Identifiable object with same id {} is already stored in this store"
+                .format(x.id))
         self._backend[x.id] = x
 
     def discard(self, x: _IT) -> None:
         if self._backend.get(x.id) is x:
             del self._backend[x.id]
 
+    def update_referable(self,
```

Comment:
The functions should be in `AbstractObjectStore` not in the concrete implementation of a `DictObjectStore`. 

Since they do not depend on the `ObjectStore`-instance, I would decorate them as `@classmethod`.